src/core/polyhedral/cuda/mapped_scop.cc

```diff
@@ -329,10 +316,8 @@ detail::ScheduleTree* MappedScop::separateReduction(detail::ScheduleTree* st) {
 size_t MappedScop::mapToThreads(detail::ScheduleTree* band, size_t nInner) {
   using namespace tc::polyhedral::detail;
 
-  if (nInner >= numThreads.view.size()) {
-    return nInner;
-  }
-  if (reductionBandUpdates_.count(band) == 1) {
+  bool mapReduction = reductionBandUpdates_.count(band) == 1;
+  if (mapReduction) {
     // A reduction is assumed to get mapped to threadIdx.x
     if (nInner != 0) {
```

Comment:
This test can then be removed.

Reply:
It will be removed in the next commit.  The first one only deals with separating the reduction band without changing how the threads are mapped in general.